pallets/dispatcher/src/lib.rs

```diff
@@ -169,6 +177,37 @@ pub mod pallet {
 			AaveManagerAccount::<T>::put(account);
 			Ok(())
 		}
+
+		/// Dispatch a call with extra gas.
+		///
+		/// This allows executing calls with additional weight (gas) limit.
+		/// The call will fail if it would consume more weight than the specified limit.
+		#[pallet::call_index(3)]
+		#[pallet::weight({
+			let call_weight = call.get_dispatch_info().weight;
+			let call_len = call.encoded_size() as u32;
+			let gas_weight = T::GasWeightMapping::gas_to_weight(*extra_gas, true);
+			T::WeightInfo::dispatch_with_extra_gas(call_len)
+				.saturating_add(call_weight)
+				.saturating_add(gas_weight)
+		})]
+		pub fn dispatch_with_extra_gas(
+			origin: OriginFor<T>,
+			call: Box<<T as Config>::RuntimeCall>,
+			extra_gas: u64,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			// Store the gas limit for this account
+			ExtraGas::<T>::set(extra_gas);
+
+			// Dispatch the call
+			let (result, _) = Self::do_dispatch(who.clone(), *call);
```

Comment:
please test that this actually returns correct weight back and that you pay for extra gas and that unused extra gas is refunded as weight. seems to me that when you do it this way you dont actually pay for extra gas as inner call ahs no notion of it